packages/fxa-auth-server/lib/routes/subscriptions.js

```diff
@@ -18,13 +18,20 @@ module.exports = (log, db, config, customs, push, oauthdb, subhub) => {
   const SUBSCRIPTIONS_MANAGEMENT_SCOPE =
     'https://identity.mozilla.com/account/subscriptions';
 
-  function handleAuth(auth) {
+  async function handleAuth(auth, fetchEmail = false) {
     const scope = ScopeSet.fromArray(auth.credentials.scope);
     if (! scope.contains(SUBSCRIPTIONS_MANAGEMENT_SCOPE)) {
       throw error.invalidScopes('Invalid authentication scope in token');
     }
-    const { user, email } = auth.credentials;
-    return { uid: user, email };
+    const { user: uid } = auth.credentials;
+    let email;
+    if (! fetchEmail) {
+      ({ email } = auth.credentials);
+    } else {
+      const account = await db.account(uid);
```

Comment:
Kind of scratching my head at this change, because I thought both sessionToken and oauth token auth should come with an email address already populated. But, when I was actually trying this stuff out end-to-end, the email address was null. This seemed like the only way to actually get a hold of the email address.

Reply:
Huh, I thought they should too...

Reply:
Think I might file a follow-up issue about this - it works for now, but maybe there's something else screwy happening here.